### ISSUE TITLE
Node-specific setup steps should run automatically as nodes are admitted to the cluster

### ISSUE THREAD

**Issue by celskeggs** [sipb/homeworld#172]
Instead of these steps being performed manually:

    $ spire seq core
    $ spire seq registry
    $ spire seq flannel
    $ spire seq dns-addon

We should be able to just set a configuration option in setup.yaml:

    include-homeworld-registry: true
    
(to say that, yes, the registry *should* be included.)

And then everything else should occur automatically due to nodes being installed and automatic scripts being run.

This will accomplish two things:

 * Easier to set up a cluster
 * Allow adding nodes to an existing cluster.

**Comment by celskeggs** [comment]
The basic commands that need equivalents, which are the unpacking of the `seq` commands from above:

    $ spire infra install-packages
    $ spire setup services
    $ spire verify etcd
    $ spire verify kubernetes
       (the following three commands only if a registry is being set up)
    $ spire setup dns-bootstrap
    $ spire setup bootstrap-registry
    $ spire verify aci-pull
       (end)
    $ spire deploy flannel
    $ spire verify flannel-run
    $ spire verify flannel-ping
    $ spire deploy dns-addon
    $ spire verify dns-addon-run
    $ spire verify dns-addon-ping

There are a few categories of these commands:

  * verification commands, which should probably be converted to monitoring scripts, so that instead of running these directly, it's more of a "let's watch a dashboard to see if all the lights turn green" process
  * kubernetes deployment commands, which might need to be kicked off by a single manual command.
  * configuration that should be able to be done by the ISO, like setting up dns bootstrapping and installing most packages. (homeworld-services can be installed everywhere, since it doesn't have any install-triggered scripts, but the bootstrap registry package can't.)
  * commands that are supervisor-only, and should be able to be performed while setting up the supervisor node (i.e. `spire setup bootstrap-registry`, installing the packages for the bootstrap registry on the supervisor node)
  * commands that might require more complicated machinery to automatically perform, like `spire setup services`

**Comment by celskeggs** [comment]
New ordering (I'm ignoring registry setup checking, because honestly it's not worth the additional effort for that at this time):

      # equivalents will be moved to during ISO installation
    $ spire infra install-packages  (for everything except homeworld-bootstrap-registry)
    $ spire setup dns-bootstrap
      # done during supervisor setup
    $ spire infra install-packages  (for homeworld-bootstrap-registry)
    $ spire setup bootstrap-registry
      # needs more complicated replacement
    $ spire setup services
    $ spire deploy flannel
    $ spire deploy dns-addon
      # will be handled via monitoring
    $ spire verify etcd
    $ spire verify kubernetes
    $ spire verify aci-pull
    $ spire verify flannel-run
    $ spire verify flannel-ping
    $ spire verify dns-addon-run
    $ spire verify dns-addon-ping

**Comment by celskeggs** [comment]
Latest ordering and comments:

```
  # implemented by monitoring
$ spire verify online
  # being moved to ISO installation
$ spire setup dns-bootstrap   # being worked on; see #188
  # now implemented by monitoring
$ spire verify etcd
$ spire verify kubernetes
  # needs more complicated replacement
$ spire deploy flannel
$ spire deploy dns-addon
$ spire deploy flannel-monitor
$ spire deploy dns-monitor
  # now implemented by monitoring
$ spire verify aci-pull
$ spire verify flannel
$ spire verify dns-addon
```

**Comment by celskeggs** [comment]
Since #229 was merged, the only remaining step is #188; every other step is run automatically as nodes are admitted -- or moved to the supervisor init phase. This issue is now redundant, and can be closed.
